### PR TITLE
JS pre OOP

### PR DESCRIPTION
**Pull request by ustymchyk**
### Building a Tiny JS World

[Demo](https://ustymchyk.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/ustymchyk/a-tiny-JS-world/tree/populate-world/index.js)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/ustymchyk/js-pre-oop/index.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+[dog, cat, woman, man].forEach(obj => print(convertToString(obj)));
+
+function convertToString(obj) {
+  return Object.values(obj).flat().join('; ');
```
Comment (OleksiyRudenko) [inline submissions/ustymchyk/js-pre-oop/index.js:69]: `Object` methods do not guarantee any particular order of keys/values. That means under different browser JS engine or under different BE JS engine the result can be different.
And here particular order is important to interpret the output properly.
Solution: list properties explicitly. Hint: `['prop1'].map(propName => obj[propName])`

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@ustymchyk well done.
Whenever a particular order of anything is required `Object` methods are bad friends.

**Comment by ustymchyk** [comment]
done

**Review by OleksiyRudenko — APPROVED** [review]
@ustymchyk nice job!
Approved.
However, based on the fact that you basically completed the entire phase I believe you could do better.

I mean the code could be DRYer in the fragment below

```
const convertHumanToString = (obj) => {
  return ['species', 'name', 'gender', 'saying', 'legs', 'hands', 'friends']
    .map(property => obj[property]).join('; ');
};

const convertPatToString = (obj) => {
  return ['species', 'name', 'gender', 'saying', 'legs', 'friends']
    .map(property => obj[property]).join('; ');
};
```

**Pull request merged**